### PR TITLE
Sweets/cleanup

### PR DESCRIPTION
**Pull request by sweetmantech**
# Cleanup before sunday shipment
- update NATSPEC docs
- remove `useXHook` for Transfer Hooks - this logic can be taken care of in external hooks. Keep our hooks simple and light weight.

### CODE REVIEW

File: src/ERC721ACH.sol [hunk 1]
```diff
@@ -1,36 +1,34 @@
...
-    * Only the contract owner can set or update these hooks.
-    */
+     * @dev This mapping associates hook types with their corresponding contract addresses.
+     * Each hook type can be associated with a contract that implements the hook's logic.
+     * Only the contract owner can set or change these hooks.
+     */
     mapping(HookType => address) public hooks;
 
-
-    event UpdatedHook(address indexed setter, HookType hookType, address indexed hookAddress);
```
Comment (SweetmanTech) [inline src/ERC721ACH.sol:28]: moved this event into the interface

**Comment by ahmedali8** [comment]
useXHook should be removed imo as it's an extra external call just to check if we want to run the hook and we're already checking that by the hook's valid address.

**Review by crazzywizard — APPROVED** [review]

File: src/ERC721ACH.sol [hunk 2]
```diff
@@ -99,23 +99,20 @@ contract ERC721ACH is IERC721ACH, ERC721AC {
      */
     function ownerOf(
         uint256 tokenId
-    ) public view virtual override returns (address) {
-        address owner;
+    ) public view virtual override returns (address owner) {
```
Comment (ahmedali8) [inline src/ERC721ACH.sol:102]: Please check my PR again https://github.com/defientco/ERC721H/pull/16

you have undo my changes here which is incorrect

**Comment by ahmedali8** [comment]
Looks good to me!

**Review by crazzywizard — APPROVED** [review]

**Review by ahmedali8 — APPROVED** [review]

**Pull request merged**